frontend/client-list/client-list.component.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { useFullWidth } from "../navbar/use-full-width.hook";
+import PageHeader from "../page-header.component";
+import ClientsTableToolbar from "./toolbar/clients-table-toolbar.component";
+import ClientsTable from "./table/clients-table.component";
+import easyFetch from "../util/easy-fetch";
+import { SingleClient } from "../view-edit-client/view-client.component";
+import { useQueryParamState } from "../util/use-query-param-state.hook";
+
+const pageSize = 100;
+
+export default function ClientList(props: ClientListProps) {
+  useFullWidth();
+  const [clientApiData, setClientApiData] = React.useState<ClientApiData>({
+    numClients: 150,
+    clients: []
+  });
+  const [fetchingClients, setFetchingClients] = React.useState(true);
+  const [page, setPage] = useQueryParamState("page", "1", Number);
+
+  React.useEffect(() => {
+    const lastPage = Math.ceil(clientApiData.numClients / pageSize);
+
+    if (typeof page !== "number" || isNaN(page)) {
+      setPage(1);
+    } else if (page <= 0) {
+      setPage(1);
+    } else if (page > lastPage) {
+      setPage(lastPage);
+    }
+  }, [page, clientApiData]);
+
+  // React.useEffect(() => {
```

Comment:
Commented out until we're actually integrating with the API, or at least mocking its response with dummy data.